### ISSUE TITLE
Enums with numeric values defined in enumValues are not generated correctly (when enumsAsConst=true)

### ISSUE THREAD

**Issue by joshboley** [dotansimha/graphql-code-generator#4870]
**Describe the bug**
This bug was fixed in https://github.com/dotansimha/graphql-code-generator/issues/3137.  However, it was only fixed if `enumAsConst` is NOT set to `true`.  When `enumAsConst=true`, the integer 0 is not converted, but is instead set to the string matching the key.

**To Reproduce**

<!--
#############
#############
🛑 STOP! 🛑 STOP! 🛑 STOP! 🛑 STOP! 🛑 STOP! 🛑 STOP! 🛑 STOP! 🛑 STOP! 🛑 STOP! 🛑 STOP! 🛑 STOP! 

PLEASE READ THE FOLLOWING: 
You can use this template to create a live sandbox of the issue: 
https://codesandbox.io/s/github/dotansimha/graphql-code-generator-issue-sandbox-template
Make sure sure fork this template, and run `yarn generate` in the terminal. 
Please make sure the Codegen and plugins version under `package.json` matches yours. 
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~~~~~~~~ ISSUES WITH A LIVE REPRODUCTION WILL BE ANSWERED QUICKLY ~~~~~~~
~~~~~~ ISSUES WITHOUT REPRODUCTION MIGHT GET CLOSED WITHOUT RESPONSE ~~~~~
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#############
#############
-->

https://codesandbox.io/s/festive-jennings-83lwx

1. My GraphQL schema:

```graphql
enum MyEnum {
    Value1
    Value2
}
```

2. My GraphQL operations:

N/A, this part is fine.

3. My `codegen.yml` config file:

```yml
schema: schema.graphql
documents: document.graphql
generates:
  types.ts:
    plugins:
      - typescript
      - typescript-operations
    config:
      enumsAsConst: true
      enumValues:
        MyEnum:
          Value1: 0
          Value2: 1
```

**Expected behavior**
<!-- A clear and concise description of what you expected to happen. -->

Expecting the output type to look like this:
```
export const MyEnum = {
  Value1: 0,
  Value2: 1
} as const;

export type MyEnum = typeof MyEnum[keyof typeof MyEnum];
```

Instead, it looks like this:
```
export const MyEnum = {
  Value1: 'Value1',
  Value2: 1
} as const;

export type MyEnum = typeof MyEnum[keyof typeof MyEnum];
```

**Environment:**

- OS: Linux/Mac
- `@graphql-codegen/...`:  (also impacts 1.17.4, other versions not tested)
```
"dependencies": {
    "@graphql-codegen/add": "1.17.0",
    "@graphql-codegen/cli": "1.17.0",
    "@graphql-codegen/typescript": "1.17.0",
    "@graphql-codegen/typescript-operations": "1.17.0",
    "graphql": "15.3.0"
  }
```
- NodeJS: Tested with v14.8.0 and v10.21.0

**Additional context**
<!-- Add any other context about the problem here. -->

**Comment by joshboley** [comment]
Thanks @borremosch for fixing this so quickly! @dotansimha what's your release cadence?  Just curious when this might be released.

**Comment by dotansimha** [comment]
Fixed in `@graphql-codegen/typescript@1.17.11`
